src/player.cpp

```diff
@@ -11998,6 +11998,10 @@ void player::do_read( item *book )
     auto reading = book->type->book.get();
     const skill_id &skill = reading->skill;
 
+    if( reading == NULL ) {
```

Comment:
If reading is null, the line above (`reading->skill`) already causes undefined behavior.